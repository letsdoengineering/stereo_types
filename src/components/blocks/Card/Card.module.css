.card {
    border: 1px solid var(--outline-gray);
}

.card--padded {
    padding: var(--standard-padding)
}