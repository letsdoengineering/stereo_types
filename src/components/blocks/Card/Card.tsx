import React, { AllHTMLAttributes } from 'react'
import * as cardStyles from './Card.module.css'
import getClassNames from '../../../utils/get-class-names'

type CardProps = AllHTMLAttributes<HTMLDivElement> & {
    isPadded?: boolean
    className?: string
    children?: React.ReactNode
}

const Card: React.FC<CardProps> = ({ isPadded = false, className, children }) => {
    const classNames = getClassNames({
        defaultClasses: [`${cardStyles.card}`],
        className,
        conditionalClasses: {
            [`${cardStyles.cardPadded}`]: isPadded,
        },
    })

    return <div className={classNames}>{children}</div>
}

export default Card
