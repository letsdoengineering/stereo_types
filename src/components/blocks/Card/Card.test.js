import Card from './Card'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

describe('Card', () => {
  it('has not changed the JSX layout of the page from previous snapshot - isPadded=true', () => {
    const tree = renderer.create(<Card isPadded />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('has not changed the JSX layout of the page from previous snapshot - isPadded=false', () => {
    const tree = renderer.create(<Card />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<Card isPadded>Some text</Card>)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
