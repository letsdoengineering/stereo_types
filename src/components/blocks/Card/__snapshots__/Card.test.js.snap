// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card has not changed the JSX layout of the page from previous snapshot - isPadded=false 1`] = `
<div
  className="card"
/>
`;

exports[`Card has not changed the JSX layout of the page from previous snapshot - isPadded=true 1`] = `
<div
  className="card cardPadded"
/>
`;
